test/e2e/simple.go

```diff
@@ -51,11 +50,14 @@ func E2ESimple(logger *log.Logger) error {
 	defer cancel()
 	opts := txsim.DefaultOptions().WithSeed(seed).SuppressLogs()
 	err = txsim.Run(ctx, testNet.GRPCEndpoints()[0], kr, encCfg, opts, sequences...)
-
-	if !errors.Is(err, context.DeadlineExceeded) {
-		return fmt.Errorf("expected context.DeadlineExceeded, got %w", err)
```

Comment:
I don't understand this. It failed me, as the error returned is not `context.DeadlineExceeded`. Commented it out for now.

Reply:
+1 on not understanding this. I created https://github.com/celestiaorg/celestia-app/issues/3380 which we can close if we end up deleting these lines.

Reply:
What error did you get? Was it context cancelled?

Reply:
This should always return an error

Reply:
cc @smuu (just so we don't forget this)

Reply:
I don't get any error when running txsim.
When having the code to expect `context.DeadlineExceeded`, the test fails with this specified error.

Reply:
@cmwaters What was the logic for you to add this check?